price_pusher/src/sui/sui.ts

```diff
@@ -244,34 +288,199 @@ export class SuiPricePusher implements IPricePusher {
     return tx;
   }
 
-  /** Send every transaction in txs sequentially, returning when all transactions have completed. */
-  private async sendTransactionBlocks(txs: TransactionBlock[]): Promise<void> {
-    for (const tx of txs) {
-      try {
-        const result = await this.signer.signAndExecuteTransactionBlock({
-          transactionBlock: tx,
-          options: {
-            showInput: true,
-            showEffects: true,
-            showEvents: true,
-            showObjectChanges: true,
-            showBalanceChanges: true,
-          },
-        });
-
-        console.log(
-          "Successfully updated price with transaction digest ",
-          result.digest
+  /** Send every transaction in txs in parallel, returning when all transactions have completed. */
+  private async sendTransactionBlocks(
+    txs: TransactionBlock[]
+  ): Promise<void[]> {
+    return Promise.all(txs.map((tx) => this.sendTransactionBlock(tx)));
+  }
+
+  /** Send a single transaction block using a gas coin from the pool. */
+  private async sendTransactionBlock(tx: TransactionBlock): Promise<void> {
+    const gasObject = this.gasPool.shift();
+    if (gasObject === undefined) {
+      console.warn("No available gas coin. Skipping push.");
+      return;
+    }
+
+    let nextGasObject: SuiObjectRef | undefined = undefined;
+    try {
+      tx.setGasPayment([gasObject]);
+      tx.setGasBudget(this.gasBudget);
+      const result = await this.signer.signAndExecuteTransactionBlock({
+        transactionBlock: tx,
+        options: {
+          showEffects: true,
+        },
+      });
+
+      nextGasObject = getTransactionEffects(result)
```

Comment:
I'm not understanding it correctly, how nextGasObject couldn't be the old gasObject here?